src/raw/http_util/uri.rs

```diff
@@ -44,6 +48,21 @@ pub fn percent_encode_path(path: &str) -> String {
     utf8_percent_encode(path, &PATH_ENCODE_SET).to_string()
 }
 
+/// check endpoint of service.
+///
+/// If endpoint is empty, return error.
+pub fn check_endpoint(endpoint: &String, schema: Scheme) -> Result<String> {
```

Comment:
**request**: `Scheme` and `Builder::build` are part of error contexts and they should not be handled by a util function.

**question**: Currently logic is simple that I don't think it worth a seperate util function.